### PR TITLE
[github-org] early exit pr-check dry-runs

### PR DESCRIPTION
**Pull request by geoberle**
part of https://issues.redhat.com/browse/APPSRE-6098

### CODE REVIEW

File: reconcile/github_org.py [hunk 1]
```diff
@@ -499,3 +499,10 @@ def run(dry_run):
     )
 
     runner.run()
+
+
+def early_exit_desired_state(*args, **kwargs) -> dict[str, Any]:
+    return {
+        "github_org_config": get_config(),
```
Comment (maorfr) [inline reconcile/github_org.py:506]: why isn't this enough? https://github.com/app-sre/qontract-reconcile/blob/4c5749bb25ebb08741c3f1579a259801c5f53fad/reconcile/github_org.py#L95

File: reconcile/github_org.py [hunk 2]
```diff
@@ -499,3 +499,10 @@ def run(dry_run):
...
+        "github_org_members": fetch_desired_state().dump(),
```
Comment (maorfr) [inline reconcile/github_org.py:507]: wouldn't you prefer to replace this with `permissions` and `clusters`?

**Comment by maorfr** [comment]
both these comments are nits, but i'm trying to help us understand what is the bare minimum we need to do in early exit.

File: reconcile/github_org.py (hunk 1 above)
Comment (maorfr) [reply reconcile/github_org.py:506]: do you mean managed teams?

Comment (maorfr) [reply reconcile/github_org.py:506]: if so, the list of managed teams is a part of the query: https://github.com/app-sre/qontract-reconcile/blob/4c5749bb25ebb08741c3f1579a259801c5f53fad/reconcile/github_org.py#L32

Comment (geoberle) [reply reconcile/github_org.py:506]: yes sry i meant `managedTeams`. the query you posted is exactly used by `get_config()`

Comment (geoberle) [reply reconcile/github_org.py:506]: maybe i misunderstood you question. let me revisit

Comment (geoberle) [reply reconcile/github_org.py:506]: ah ok i get it. you mean "why the additional processing of `get_config` if the core information required is get result of the query. good point. i will refactor it

Comment (geoberle) [reply reconcile/github_org.py]: see [34feef4](https://github.com/app-sre/qontract-reconcile/pull/2698/commits/34feef40903e8a03583bbeb0ddcef7fc7dae39d4)

File: reconcile/github_org.py (hunk 2 above)
Comment (geoberle) [reply reconcile/github_org.py:511]: queried `clusters` are intertwined with `openshift_users.fetch_desired_state(oc_map=None)` so this processing done in `fetch_desired_state` seems to meaningful for getting the picture about the desired state.

**Review by maorfr — APPROVED** [review]

**Pull request merged**